src/Timeout/Timeout.php

```diff
@@ -66,7 +67,23 @@ private static function init(): void
         // Setup a handler for SIGALRM that throws an exception
         // This will interrupt any running PHP code, including `sleep()` or code stuck waiting for I/O.
         pcntl_signal(SIGALRM, function (): void {
-            throw new LambdaTimeout('Maximum AWS Lambda execution time reached');
+            if (Timeout::$stackTrace !== null) {
+                // we already thrown an exception.
+                error_log('Lambda timed out');
+                error_log((new LambdaTimeout())->getTraceAsString());
+                error_log('Original stack trace');
+                error_log(Timeout::$stackTrace);
+                Timeout::$stackTrace = null;
+                exit(1);
```

Comment:
Should we force a `500` status code before exit?

That would only be taken into account in the FPM layer, so that wouldn't really impact the `php-XX` layer.

Reply:
How? 
And wouldn't that effect my `AWS::ApiGateway::GatewayResponse` that triggers on timeouts or 500?

Reply:
```php
http_response_code(500);
```

> And wouldn't that effect my AWS::ApiGateway::GatewayResponse that triggers on timeouts or 500?

🤔 not sure about that, maybe. But in any case, the same would happen if your app returns a pretty 500 error page.

Does that make sense? What I mean is that, when using FPM, there would be no difference between:

- Symfony catches any exception and displays an error page with a 500 status code
- Bref timeout kicks in and forces a 500 status code

In both cases, API Gateway would behave the same way.

Reply:
Oh, no that is not true. 

> Symfony catches any exception and displays an error page with a 500 status code

This will show you a nice error page. API Gateway does nothing

> Bref timeout kicks in and forces a 500 status code

No content(HTML) is provided. I would like the custom error page in API Gateway to be used. 

-------

Im not sure if `http_response_code(500)` would make any difference. 


-------

Also note. We are not sure it is a HTTP context. It could be any other event.. So Im not sure about sending a HTTP status code.